src/os/bluestore/BlueFS.h

```diff
@@ -363,9 +441,30 @@ class BlueFS {
       buffer_appender.append_zero(len);
     }
 
+    void append(uint64_t value) {
+      uint64_t l0 = get_buffer_length();
+      ceph_assert(l0 + sizeof(value) <= std::numeric_limits<unsigned>::max());
```

Comment:
This assert is strange.
Either unsigned is 64 bit, then it will never assert (l0 + sizeof(value) will wrap around)
OR
there is no need for uint64_t in bluefs_wal_header_t::flush_length.

Reply:
ah true, will use uint64_t in assert